src/allmydata/web/root.py

```diff
@@ -182,9 +183,44 @@ def data_import_path(self, ctx, data):
     def render_my_nodeid(self, ctx, data):
         tubid_s = "TubID: "+self.client.get_long_tubid()
         return T.td(title=tubid_s)[self.client.get_long_nodeid()]
+
     def data_my_nickname(self, ctx, data):
         return self.client.nickname
 
+    def renderHTTP(self, ctx):
+        req = inevow.IRequest(ctx)
+        t = get_arg(req, "t")
+        if t == "json":
+            return self.json_welcome(ctx)
+        return rend.Page.renderHTTP(self, ctx)
+
+    def json_welcome(self, ctx):
+        inevow.IRequest(ctx).setHeader("content-type", "text/plain")
+        intro_summaries = [s.summary for s in self.client.introducer_connection_statuses()]
+        sb = self.client.get_storage_broker()
+        servers = {}
+        for server in sb.get_known_servers():
+            server_id = server.get_serverid()
+            servers[server_id] = {}
+            servers[server_id]["connection_status"] = server.get_connection_status().summary
+            servers[server_id]["available_space"] = server.get_available_space()
+            servers[server_id]["nickname"] = server.get_nickname()
+            if server.get_version() is not None:
+                servers[server_id]["version"] = server.get_version()["application-version"]
+            else:
+                servers[server_id]["version"] = ""
+            if server.rref is not None:
+                servers[server_id]["last_received_data"] = server.rref.getDataLastReceivedAt()
+            else:
+                servers[server_id]["last_received_data"] = ""
+            data = {
```

Comment:
(action required)

The indentation for the body of this `dict` is weird.  Can you normalize it?